airflow/models/dagcode.py

```diff
@@ -194,4 +194,9 @@ def dag_fileloc_hash(full_filepath: str) -> int:
         import hashlib
 
         # Only 7 bytes because MySQL BigInteger can hold only 8 bytes (signed).
-        return struct.unpack(">Q", hashlib.sha1(full_filepath.encode("utf-8")).digest()[-8:])[0] >> 8
+        return (
+            struct.unpack(
+                ">Q", hashlib.sha1(full_filepath.encode("utf-8"), usedforsecurity=False).digest()[-8:]
+            )[0]
+            >> 8
+        )
```

Comment:
Maybe adding a small unit test would be nice? (sanity check for the next time the algorithm changes)

Reply:
We already have CodeQL scanning and detecting those. I think it's unlkely we will ever remove the flag, so it's not really prone to regression, what is more worrying is if we start using it elsewhere - but for that CodeQL scanning **should** in principle detect such future usages.